docs/source/how_to/how_to_specify_algorithm_and_algo_options.md

```diff
@@ -1,46 +1,153 @@
+
+
+om.minimize(
+    fun=fun,
+    params=np.arange(3),
+    algorithm=om.algos.scipy_neldermead,
+)
+```
+
+The algorithm is selected by passing an algorithm class. This class is usually not
+imported manually, but discovered using `om.algos`. After typing `om.algos.`, your
+editor will show you all algorithms you can choose from.
+
+## Configuring an algorithm
+
+To configure an algorithm with advanced options, you can create an instance of the
+class:
+
+```python
+algo = om.algos.scipy_neldermead(
+    stopping_maxiter=100,
+    adaptive=True,
+)
+
+om.minimize(
+    fun=fun,
+    params=np.arange(3),
+    algorithm=algo,
+)
+```
+
+Again, you can use your editor's autocomplete to discover all options that your chosen
+algorithm supports. When the instance is created, the types and values of all options
+are checked. Should you make a mistake, you will get an error before you run your
+optimization.
+
+## Advanced autocomplete in action
+
+Assume you need a gradient-free optimizer that supports bounds on the parameters.
+Moreover, you have a fixed computational budget, so you want to set stopping options.
+
+If you type `om.algos.`, your editor will show you all available optimizers and a list
+of categories you can use to filter the results. In our case, we select `GradientFree`
+and `Bounded`, and we could do that in any order we want.
+
+![autocomplete_1](../_static/images/autocomplete_1.png)
+
+After selecting one of the displayed algorithms, in our case `scipy_neldermead`, the
+editor shows all tuning parameters of that optimizer. If you start to type `stopping`,
+you will see all stopping criteria that are available.
 
-To make it easier to understand which aspect of the optimization is influenced by an
-option, we group them with prefixes. For example, the name of all convergence criteria
-starts with `"convergence."`. In general, the prefix is separated from the option name
-by a dot.
+![autocomplete_2](../_static/images/autocomplete_2.png)
 
-Which options are supported, depends on the algorithm you selected and is documented in
-{ref}`list_of_algorithms`.
+## Modifying an algorithm
 
-An example could look like this:
+Given an algorithm, you can easily create a **modified copy** by using the `with_option`
+method.
 
 ```python
-algo_options = {
-    "trustregion.threshold_successful": 0.2,
-    "trustregion.threshold_very_successful": 0.9,
-    "trustregion.shrinking_factor.not_successful": 0.4,
-    "trustregion.shrinking_factor.lower_radius": 0.2,
-    "trustregion.shrinking_factor.upper_radius": 0.8,
-    "convergence.noise_corrected_criterion_tolerance": 1.1,
+# using copy constructors to create variants
+base_algo = om.algorithms.fides(stopping_maxiter=1000)
+algorithms = [
+    base_algo.with_option(trustregion_initial_radius=r) for r in [0.1, 0.2, 0.5]
+]
+
+for algo in algorithms:
+    minimize(
+        fun=fun,
+        params=np.arange(3),
+        algorithm=algo,
+    )
+```
+
+:::
+:::{tab-item} The scipy way
```

Comment:
Do we accept the scipy method alias, and the scipy method names aliases already? In any case, I think this should be mentioned here. I.e., it is either slightly different or it is possible to go "full scipy." But please convince me from the opposite.

Reply:
I added a note.